include/deal.II/fe/fe_tools.h

```diff
@@ -777,6 +782,29 @@ namespace FETools
                     const DoFHandler<dim,spacedim> &dof2,
                     const ConstraintMatrix         &constraints,
                     OutVector                      &z2);
+
+  /**
+   * Same as above for external parallel VectorTypes
+   * on a parallel::distributed::Triangulation
+   */
+  template <int dim, class VectorType, int spacedim>
+  void extrapolate_parallel(const DoFHandler<dim,spacedim> &dof1,
```

Comment:
Right. If the call to `reinit` is the only one that's offending, then you can do
```
  namespace {
    void do_reinit (p::d::Vector &x,
                           ...)   // expand argument list
    {
       x.reinit (...);
    }

   template <typename VectorType>
    void do_reinit (VectorType &,
                           ...)    // may even really be an ellipsis here :-)
    {
       Assert (false, ExcMessage ("This operation is not allowed for this kind of vector."));
    }
```
It would become a bit cumbersome if you had multiple places where you need this kind of dispatch, but if it's just for the reinit call, then that's totally feasible.